dev-client/src/screens/SoilScreen/components/EditIntervalModalContent.tsx

```diff
@@ -181,3 +205,36 @@ const AddButton = ({action, isDisabled}: AddButtonProps) => {
     </Button>
   );
 };
+
+type SwitchProps = {
+  method: SoilPitMethod;
+  isRequired: boolean;
+  value: boolean;
+  updateEnabled: (newValue: boolean) => void;
+};
+
+const InputFormSwitch = ({
+  method,
+  isRequired,
+  value,
+  updateEnabled,
+}: SwitchProps) => {
+  const {t} = useTranslation();
+
+  const label = useMemo(
+    () =>
+      t(`soil.collection_method.${method}`) +
+      (isRequired ? ` (${t('general.required')})` : ''),
+    [t, method, isRequired],
+  );
```

Comment:
issue: don't concatenate strings. This won't work for RTL (and perhaps some other languages).  Perhaps something like 
```
(isRequired ? t('general.required_item_label', { item: t(`soil.collection_method.${method}`) }
            : t(`soil.collection_method.${method}`)
```

Where `required_item_label` is `{item}: (required)`.

Reply:
In some languages (RTL or not), the `required` string might need to appear in a different position in the string, i.e. `(required) blah`.  We can't assume word order is consistent.

Reply:
Got it